vnext/Shared/MemoryMappedBuffer.cpp

```diff
@@ -0,0 +1,154 @@
+#include "pch.h"
+#include "MemoryMappedBuffer.h"
+
+#include <werapi.h>
+#include <windows.h>
+
+namespace {
+
+class MemoryMappedBuffer : public facebook::jsi::Buffer {
+ public:
+  MemoryMappedBuffer(const wchar_t *const filename, uint32_t offset);
+
+  size_t size() const override;
+  const uint8_t *data() const override;
+
+ private:
+  static void fileDataDeleter(void *p) {
+    WerUnregisterMemoryBlock(p);
+    UnmapViewOfFile(p);
+  }
+
+  std::unique_ptr<void, decltype(&CloseHandle)> m_fileMapping;
+  std::unique_ptr<void, decltype(&fileDataDeleter)> m_fileData;
+  uint32_t m_size;
+  uint32_t m_offset;
+};
+
+MemoryMappedBuffer::MemoryMappedBuffer(
+    const wchar_t *const filename,
+    uint32_t offset)
+    : m_fileMapping{nullptr, &CloseHandle},
+      m_fileData{nullptr, &fileDataDeleter},
+      m_size{0},
```

Comment:
>{0} [](start = 12, length = 3)

Please make this assignment on line 24 during the declaration

Reply:
I am more than happy to make this change. Out of curiosity though, what's the advantage for the assignment syntax?

---
In reply to: [320902973](https://github.com/microsoft/react-native-windows/pull/3059#discussion_r320902973) [](ancestors = 320902973)

Reply:
I added default intializers in the class declaration.

---
In reply to: [320909429](https://github.com/microsoft/react-native-windows/pull/3059#discussion_r320909429) [](ancestors = 320909429,320902973)

Reply:
The advantage is that it's easier to track where the default values are coming from. In this case, it doesn't matter too much, but imagine when there are multiple constructors, it could be difficult to ensure that object is initialized to same values in all of them.

---
In reply to: [320991285](https://github.com/microsoft/react-native-windows/pull/3059#discussion_r320991285) [](ancestors = 320991285,320909429,320902973)